deployments/cdi/sriovdp-daemonset.yaml

```diff
@@ -0,0 +1,115 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  annotations:
+    kubernetes.io/description: |
+      This daemon set launches the SR-IOV network device plugin on each node.
+  name: sriov-device-plugin
+  namespace: kube-system
+  labels:
+    tier: node
+    app: sriovdp
+spec:
+  revisionHistoryLimit: 10
+  selector:
+    matchLabels:
+      app: sriov-device-plugin
+  template:
+    metadata:
+      creationTimestamp: null
+      labels:
+        app: sriov-device-plugin
+        component: network
+        openshift.io/component: network
+        type: infra
+    spec:
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+              - matchExpressions:
+                  - key: node-role.kubernetes.io/worker
+                    operator: In
+                    values:
+                      - ""
+      containers:
+        - args:
+            - --log-level=10
+            - --resource-prefix=example.com
+            - --config-file=/etc/pcidp/$(NODE_NAME)
+            - --use-cdi
+          env:
+            - name: NODE_NAME
+              valueFrom:
+                fieldRef:
+                  apiVersion: v1
+                  fieldPath: spec.nodeName
+          image: ghcr.io/k8snetworkplumbingwg/sriov-network-device-plugin:latest-amd64
+          imagePullPolicy: IfNotPresent
+          name: sriov-device-plugin
+          resources:
+            requests:
+              cpu: 10m
+              memory: 50Mi
+          securityContext:
+            privileged: true
+          terminationMessagePath: /dev/termination-log
+          terminationMessagePolicy: File
+          volumeMounts:
+            - mountPath: /var/lib/kubelet/
+              name: devicesock
+            - mountPath: /etc/pcidp/
+              name: config-volume
+              readOnly: true
+            - mountPath: /var/run/k8s.cni.cncf.io/devinfo/dp
+              name: device-info
+            - mountPath: /etc/cdi/
+              name: default-cdi
+            - mountPath: /var/run/cdi
+              name: dynamic-cdi
+      dnsPolicy: ClusterFirst
+      hostNetwork: true
+      imagePullSecrets:
+        - name: ngc-image-secret
+      nodeSelector:
+        beta.kubernetes.io/os: linux
+        node-role.kubernetes.io/worker: ""
+      priorityClassName: system-node-critical
+      restartPolicy: Always
+      schedulerName: default-scheduler
+      securityContext: {}
+      serviceAccount: sriov-device-plugin
+      serviceAccountName: sriov-device-plugin
+      terminationGracePeriodSeconds: 30
+      tolerations:
```

Comment:
can you define this yaml like what we currently have under deployments  (with the added mounts and flag ofc) ?